moveit_core/robot_model/src/robot_model.cpp

```diff
@@ -1408,5 +1408,50 @@ void RobotModel::computeFixedTransforms(const LinkModel* link, const Eigen::Isom
                              transform * link->getJointOriginTransform(), associated_transforms);
 }
 
+// Calculating Hash for RobotModel
+namespace std
+{
+	template<> struct hash<RobotModel>
+	{
+		std::size_t operator()(RobotModel const& r) const noexcept 
+    {
+			std::size_t h =0;
+      boost::hash_combine(h, r.model_name_); 
+      boost::hash_combine(h, r.model_frame_);
+
+
+      /**
+       * RobotModel consists of 2: Joint and Link
+       * Additionally it has 1) Indexing for Joint and Link 2) Groups
+       * Groups would be needed for EndEffectors which comes from SRDF (yet to look at JointModelGroups)
+       */
+
+      for (LinkModel* link_model : link_model_vector_) {
```

Comment:
isn't this unsorted? Maybe we could use the `joint_model_map_` instead and sort based on key?

Reply:
I wasn't sure for either of them (`joint_model_vector` or `link_model_vector_`) In `RobotModel::buildRecursive` it seems like it is ordered based on their "state"  (I am still not sure what is state!) I am kind of inclined on sorting based on name, something along the line of `std::sort(joint_model_vector_.begin(), join_model_vector_.end(), OrderJointModelByName)`. We do have `OrderJointsByIndex` but that's of course by index that the RobotModel computes. For hash we are sure that we want the `name` of joints/link to match so we can be sure with `OrderJointModelByName`. Thoughts?